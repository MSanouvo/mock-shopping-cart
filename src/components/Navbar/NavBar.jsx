import { Link } from "react-router-dom";
import styles from "./NavBar.module.css";
import Cart from "../../assets/cart-icon-white.svg";

function NavBar() {
  return (
    <div className={styles.navBar}>
      <div className={styles.home}>
        <Link to="/">
          <h1>Home</h1>
        </Link>
      </div>
      <div className={styles.links}>
        <Link to="/store">
          <h2>Shop</h2>
        </Link>
        <Link to="/cart">
          <button className={styles.button}>
            <img className={styles.icon} src={Cart} alt="cart icon" />
          </button>
        </Link>
      </div>
    </div>
  );
}

export default NavBar;
