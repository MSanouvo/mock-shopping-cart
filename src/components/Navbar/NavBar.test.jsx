import { describe, it, expect } from "vitest";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "./NavBar";

describe("NavBar Component", () => {
    // it("renders navbar", () => {
    //     const {nav} = render (<NavBar />)
    //     expect(nav).toMatchSnapshot()
    // })

    it("renders Home Header", () => {
        render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        )
        expect(screen.getByRole("heading", {level: 1})).toHaveTextContent("Home")
    })

    it("renders Shop and Cart Headers", () => {
        render(
            <MemoryRouter>
                <NavBar />
            </MemoryRouter>
        )
        expect(screen.getByRole("heading", {name: "Shop"})).toBeInTheDocument()
        expect(screen.getByAltText("cart icon")).toBeInTheDocument()
    })

    //Router Tests goes here
})