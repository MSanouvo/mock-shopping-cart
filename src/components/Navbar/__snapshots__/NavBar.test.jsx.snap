// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar Component > renders navbar 1`] = `undefined`;
